test/challtestsrv/challenge-servers.go

```diff
@@ -0,0 +1,123 @@
+package challtestsrv
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"sync"
+)
+
+// ChallSrv is a multi-purpose challenge server. Each ChallSrv may have one or
+// more ACME challenges it provides servers for.
+type ChallSrv struct {
+	log *log.Logger
+	// Shutdown is a channel used to request the challenge server cleanly shut down
+	shutdown chan bool
+
+	// httpOneAddrs are the HTTP-01 challenge server bind address(es)/port(s). If
+	// none are specified no HTTP-01 challenge server is run. If multiple are
+	// specified an HTTP-01 challenge response server will be bound to each
+	// address.
+	httpOneAddrs []string
+	// hoMu is a RWMutex used to control concurrent updates to the HTTP-01
```

Comment:
Is it necessary to have two different mutexes? Seems like contention should be low enough to just have a single mutex for the whole `ChallSrv`.

Reply:
wfm